src/main.cpp

```diff
@@ -116,14 +116,21 @@ namespace rhost {
 
     void set_memory_limit() {
 #ifdef WIN32
-        ULONGLONG total;
-        if (!GetPhysicallyInstalledSystemMemory(&total)) {
+        MEMORYSTATUSEX ms = {};
+        ms.dwLength = sizeof ms;
+        if (!GlobalMemoryStatusEx(&ms)) {
+            logf("Couldn't set R memory limit - GlobalMemoryStatusEx failed with GetLastError=%u\n", GetLastError());
             return;
         }
 
+        double memory_limit = static_cast<double>(ms.ullTotalPhys / 1024 / 1024);
```

Comment:
You won't get a fractional part with this.  You need to have one of the 1024 be a double.